### PR TITLE
Major spec cleanup

### PR DESCRIPTION
**Pull request by lamawithonel**
I was going to add some new functionality, but the spec tests were a little hard to follow and a lot of the platform-dependent tests only included contexts for RedHat.  I ended up doing a pretty massive overhaul and this is the result.  Most of the work is in aeb57a5, the other commits are just in support of it.  The coverage should be just as good, if not better than it was before.  It also includes some changes for RuboCop.  Let me know if you want anything changed.

### CODE REVIEW

**Comment by jhoblitt** [comment]
Awesome!  I will take a look after travis finishes.

File: Gemfile [hunk 1]
```diff
@@ -1,25 +1,20 @@
 source 'https://rubygems.org'
 
-if puppetversion = ENV['PUPPET_GEM_VERSION']
-  gem 'puppet', puppetversion, :require => false
-else
-  gem 'puppet', :require => false
-end
+ENV['PUPPET_GEM_VERSION'] && puppetversion = ENV['PUPPET_GEM_VERSION']
+gem 'puppet', puppetversion, :require => false
```
Comment (jhoblitt) [inline Gemfile]: Why not compress this down to a one-liner?

File: spec/unit/classes/smartd_spec.rb [hunk 2]
```diff
@@ -1,541 +1,414 @@
...
+    context "on #{os}" do
+      let(:facts) do
+        facts
+      end
+
+      case facts[:osfamily]
+      when 'RedHat'
+        service_name = 'smartd'
+        smartd_conf_path = case facts[:operatingsystem]
+                           when 'RedHat', 'CentOS', 'Scientific', 'OracleLinux', 'SLC', 'OEL'
```
Comment (jhoblitt) [inline spec/unit/classes/smartd_spec.rb]: Does rubocop want this level of indentation?

File: .rubocop.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+---
```
Comment (jhoblitt) [inline .rubocop.yml]: This is great.  How about pulling in the rakefile target and adding it to travis as well?

File: Gemfile [hunk 4]
```diff
@@ -1,25 +1,20 @@
...
   # https://github.com/rspec/rspec-core/issues/1864
-  gem 'rspec', '< 3.2.0', {"platforms"=>["ruby_18"]}
+  gem 'rspec', '< 3.2.0', 'platforms' => ['ruby_18']
+  gem 'rake',                     :require => false
   gem 'puppetlabs_spec_helper',   :require => false
   gem 'puppet-lint', '>= 1.1.0',  :require => false
   gem 'puppet-syntax',            :require => false
   gem 'rspec-puppet', '~> 2.2',   :require => false
+  gem 'rspec-puppet-facts',       :require => false
   gem 'metadata-json-lint',       :require => false
```
Comment (jhoblitt) [inline Gemfile]: rubocop should be added to the Gemfile.

**Comment by jhoblitt** [comment]
Whoa!  This is fantastic.  I've made a few minor comments.

File: spec/unit/classes/smartd_spec.rb (hunk 2 above)
Comment (lamawithonel) [inline spec/unit/classes/smartd_spec.rb]: It does.  When a case statement is used to assign only one variable it wants the `var = case..` syntax, and it wants the keywords of the case statement to align.  I can disable the check if you want.  I'm not particularly fond of it myself, though I prefer to follow the defaults unless there's a good reason to differ.

File: .rubocop.yml (hunk 3 above)
Comment (lamawithonel) [inline .rubocop.yml]: I tried that initially, but it's incompatible with Ruby 1.8 and Puppet's vendor-bundled SafeYAML module.  I've added it again with some conditional logic restricting it to Ruby 2.2 and newer.  It's not as clean as the rest of the checks, but it passes the latest build.

**Comment by lamawithonel** [comment]
Have you had a chance to review this?  I addressed the issues you brought up, but I've been away for a few weeks.  I'm now using it in production with #52 if it makes any difference.

**Comment by lamawithonel** [comment]
The last merge made most of my previous changes really difficult to rebase and I pretty much have to rewrite it.  For now I'm going to scale this PR back so it can get merged more quickly.  I will open other PRs for the other changes.  This PR now just covers the Transpec deprecation cleanup and RuboCop support.

**Comment by jhoblitt** [comment]
Sorry for being asleep at the wheel here!  I'm going to merge this before I tackle the rspec rot.  Thanks!
